eth/downloader/downloader.go

```diff
@@ -941,18 +942,17 @@ func (d *Downloader) fetchHeight(p *peer) (uint64, error) {
 			}
 			return headers[0].Number.Uint64(), nil
 
+		case <-timeout:
+			glog.V(logger.Debug).Infof("%v: head header timeout", p)
+			return 0, errTimeout
+
 		case <-d.bodyCh:
```

Comment:
We can deplete the receipt and state channels here too. When fetching the remote node's chain's height, we haven't yet started any of the fetchers, so anything delivered at this point into the data channels is surely junk.